tmt/steps/__init__.py

```diff
@@ -628,6 +628,10 @@ def summary(self) -> None:
 
     def load(self) -> None:
         """ Load status and step data from the workdir """
+        if self.should_run_again and isinstance(self, tmt.steps.discover.Discover):
+            self.debug('Run discover again when reexecuting to capture changes in plans')
+            return
```

Comment:
Would it make sense to override `load()` in `tmt.steps.discover.Discover`? That way the custom handling would be performed by the class that needs it, instead of leaking into superclass.